### PR TITLE
Removed Deprecations and made usable for julia 1.0.3

### PR DESCRIPTION
**Pull request by ComputerMaestro**
I have removed the deprecations by running tests in Julia 0.7. It is now able to pass all the tests without any deprecation warnings or any errors.

### CODE REVIEW

**Comment by ComputerMaestro** [comment]
Not sure whether I have done correctly or not.
I have some thought in "core.jl" that what is scaler concept there?

File: src/namedlevels.jl [hunk 1]
```diff
@@ -23,7 +23,7 @@ function lvls(indexer, names::Vararg{Symbol})
     name_set = first.(levelname_map(indexer))
     num_set = last.(levelname_map(indexer))
     map(names) do name
-        ind = findfirst(name_set, name)
+        ind = something(findfirst(isequal(name), name_set), 0)
```
Comment (oxinabox) [inline src/namedlevels.jl]:
```suggestion
        ind = findfirst(isequal(name), name_set)
```

If this is not found then it is fine for the `ind` to be `nothing`, that will give a surficent error

**Comment by oxinabox** [comment]
> I have some thought in "core.jl" that what is scaler concept there?

`is_scalar` / and `Scalar`/`NotScalar`,  form a trait.
See https://white.ucc.asn.au/2018/10/03/Dispatch,-Traits-and-Metaprogramming-Over-Reflection.html#part-2-aslist-including-using-traits

That trait is use to decide if we need to keep going down the nested iterator
when applying functions.

**Comment by ComputerMaestro** [comment]
Thankyou, I understood the trait concept. Your post is really good.

**Comment by oxinabox** [comment]
CI is failing, 
I've not looked why but I guess `.travis.yml` needs updating
and the AppVeyor script needs to be replaced with the script from https://github.com/JuliaCI/Appveyor.jl

**Comment by ComputerMaestro** [comment]
when i did testing on windows it passes all the tests but now I saw for linux in travis.yml it was not able to pass. Same for mac.
The error is from `lvls()` at `test_namedlevels.jl` at 61 line. Instead of giving `BoundsError` [expected output to pass text] it is giving `ArgumentError`

**Comment by ComputerMaestro** [comment]
for testing purpose I changed the test at `test_namedlevels.jl` line 61[ `BoundsError` to `ArgumentError`] and now it passed tests. I don't know whether this is correct or not. Please check it.

**Comment by oxinabox** [comment]
Yes, that is the correct change.
Thanks!

**Pull request merged**